src/compute-client/src/types/sinks.rs

```diff
@@ -79,12 +79,53 @@ impl RustType<ProtoComputeSinkDesc> for ComputeSinkDesc<CollectionMetadata, mz_r
     }
 }
 
-#[derive(Arbitrary, Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
 pub enum ComputeSinkConnection<S = ()> {
     Subscribe(SubscribeSinkConnection),
     Persist(PersistSinkConnection<S>),
 }
 
+// This is just the expanded macro with an additional `S: 'static` bound on to
+// make the `boxed_union`-based strategy happy. We can remove this code and go
+// back to the derived version if we can afford to put the bound on the enum
+// definition.
```

Comment:
Putting that bound on the `enum` definition is definitely fine. It would be _extremely_ weird if we needed to support borrowed data in this type.